### PR TITLE
Correct display of cx & ccx gates and fix bug on multi-control gates in text_drawer

### LINKED ISSUE #4052: Error in text drawing of open controlled CX and CCX gates
### Information

- **Qiskit Terra version**: 0.13.0.dev0+6002304
- **Python version**: 3.7
- **Operating system**: Ubuntu 18.04

### What is the current behavior?

The control bullets are drawn closed irrespective of the provided control state in CX and CCX gates in text drawer. The same error is observed in CY gate. Other controlled and multi-controlled gates are drawn fine. See below for an instance.

### Steps to reproduce the problem

```
qc = QuantumCircuit(4)
qc.append(XGate().control(1,ctrl_state='0'),[0,1])
qc.append(XGate().control(2,ctrl_state='10'),[0,1,2])
qc.append(XGate().control(3,ctrl_state='100'),[0,1,2,3])
qc.append(YGate().control(1,ctrl_state='0'),[0,1])
qc.append(YGate().control(2,ctrl_state='10'),[0,1,2])
qc.draw('text')
```

### LINKED ISSUE #4180: Controlled Gate .draw() error: method draws incorrect gate if amount of control qubits < 3
### Information

- **Qiskit Terra version**: 0.18.0
- **Python version**: 3.7.4 (default, Aug 13 2019, 20:35:49) [GCC 7.3.0]
- **Operating system**: Linux
- *Qiskit Aqua version**: 0.7.0
That's not a "standard" versions set, expected Aqua version is 0.6.5.
Unfortunately, Controlled gates don't work at all with the standard Qiskit 0.18.0 setup.
See details at the [4175](https://github.com/Qiskit/qiskit-terra/issues/4175) bug investigation.

### What is the current behavior?
Anti-contolled gates with 1 or 2 control qubits aren't displayed correcly

### Steps to reproduce the problem
In the Jupyter notebook cell execute code
```
from qiskit import QuantumRegister, ClassicalRegister, QuantumCircuit, execute, Aer
from qiskit.extensions.standard import XGate
import qiskit.tools.jupyter
%qiskit_version_table
qreg = QuantumRegister(5) 
qc = QuantumCircuit(qreg)
control1 = XGate().control(1, ctrl_state='0')
qc.append(control1, [0, 1])
control2 = XGate().control(2, ctrl_state='00')
qc.append(control2, [0, 1, 2])
control2_2 = XGate().control(2, ctrl_state='10')
qc.append(control2_2, [0, 1, 2])
control3 = XGate().control(3, ctrl_state='010')
qc.append(control3, [0, 1, 2, 3])
qc.draw()
```

The result is
image

### What is the expected behavior?

1. q0 control symbol should be "empty circle" in the first gate;
2. q0 and q1 control symbols should be "empty circle" in the second gate;
3. one control symbols should be "empty circle" in the thrid gate;
4. fourth gate is visualized correctly.

I'm not sure where the issue is: in the visualization part, or in the gate part.

**Update 1**: I wrote an additional test. The controlled gate is formed and computed incorrectly. A code below generates a standard "CNOT" gate instead of anti-controlled not:
```
from qiskit import QuantumRegister, ClassicalRegister, QuantumCircuit, execute, Aer
from qiskit.extensions.standard import XGate

simulator = Aer.get_backend('qasm_simulator')
qreg = QuantumRegister(2) 
creg = ClassicalRegister(2)
qc = QuantumCircuit(qreg, creg)

#qc.x(qreg[0])
control1 = XGate().control(1, ctrl_state='0')
qc.append(control1, [0, 1])
qc.measure(qreg,creg)   

job = execute(qc, simulator, shots=1000)
result = job.result()
counts = result.get_counts(qc)
print("\nTotal count for 00 and 11 are:",counts)
qc.draw()
```
A circuit and results:
image
With the anti-controlled not only '01' should be measured.

### PR DESCRIPTION
**Pull request by enavarro51**
### Summary

Fixes #4052
Fixes #4180

### Details and comments

Fixed the cx and ccx gates to properly display open/closed control qubits. In the process, discovered a bug in the ctrl_state implementation where the incorrect open/closed states were sometimes being displayed. Restructured this section and tested against the maplotlib.py output. Also had to change the test for the test_controlled_composite_gate_top_bot in the test_circuit_text_drawer module.

### CODE REVIEW

File: test/python/visualization/test_circuit_text_drawer.py [hunk 1]
```diff
@@ -2105,15 +2105,15 @@ def test_controlled_composite_gate_bot(self):
...
+                              "q_0: |0>───o────",
                               "        ┌──┴───┐",
                               "q_1: |0>┤0     ├",
                               "        │      │",
                               "q_2: |0>┤1 Ghz ├",
                               "        │      │",
                               "q_3: |0>┤2     ├",
                               "        └──┬───┘",
-                              "q_4: |0>───o────",
+                              "q_4: |0>───■────",
```
Comment (ajavadia) [inline test/python/visualization/test_circuit_text_drawer.py:2116]: I don't think this change is correct. The gate is applied as 
```python
        ccghz = ghz.control(2, ctrl_state='01')
        circuit.append(ccghz, [4, 0, 1, 2, 3])
```
`ctrl_state='01' ` means open-control on the more significant control line, and closed-control on the less significant control line. The control lines are q4 and q0. q4 is more significant, so should be open.

Comment (enavarro51) [reply test/python/visualization/test_circuit_text_drawer.py:2116]: Ok. I was following the example from #3957 
```qr = QuantumRegister(4, 'q')
circuit = QuantumCircuit(qr)
circuit.append(U2Gate(pi, -5 * pi / 8).control(3, ctrl_state='100'),
               [qr[0], qr[3], qr[2], qr[1]])
circuit.draw('text')
                         
q_0: |0>────────o────────
        ┌───────┴───────┐
q_1: |0>┤ U2(pi,-5pi/8) ├
        └───────┬───────┘
q_2: |0>────────■────────
                │        
q_3: |0>────────o────────
```
and the code in each of the 3 drawers,

`        ctrl_state = "{0:b}".format(instruction.op.ctrl_state).rjust(num_ctrl_qubits, '0')[::-1]
`

which reverses ctrl_state and then applies it to each of the control qubits left to right. Is this different for the multigates?
And while we're discussing, the cz in text.py appears as an open on top and closed on the bottom, whereas in mpl it is both closed. Which way should this be?
Thanks.

Comment (ajavadia) [reply test/python/visualization/test_circuit_text_drawer.py:2116]: the output should be the same for all 3 drawers. please raise an issue if it is not. I'm not sure what they do internally, but i can say that the output that was there is correct for text drawer.

> cz in text.py appears as an open on top and closed on the bottom, whereas in mpl it is both closed

CZ with what ctrl_state? one of the controls should be open if there's a '0' in the ctrl_state.

Comment (enavarro51) [reply test/python/visualization/test_circuit_text_drawer.py:2116]: - Ok on the ghz gate as it was.
- On the cz, I was asking what the default was, but I see that cx, cy, & cz all default to ctrl_state 1, so it works fine in text.py. matplotlib.py does not address the ctrl_state for cz and needs to be fixed.
- See my comment in #4180. If we do what I suggested, I'll fix the matplotlib.py with this as well.

**Comment by 1ucian0** [comment]
I merged in the changes from #4180

**Comment by enavarro51** [comment]
Ok, @1ucian0. I guess we'll take it from here. 
@ajavadia I'll take care of the issues from my last comment above and run some tests on all this and then push the combined changes.

**Comment by enavarro51** [comment]
@ajavadia The ghz problem from above did not get fixed in this merge. I had made a fix in text.py in my version but found I needed to update matplotlib.py for the same problem. I was going to push those changes along with some fixes for rxx and cswap gates in mpl today, but Luciano merged and closed this.
I have changes ready to push. Do you want to reopen this or open a new issue to take those changes?

**Comment by kdk** [comment]
Hi @enavarro51 , please open a new PR for those changes.

**Comment by enavarro51** [comment]
@kdk Thanks. Just a procedural question. Should I continue with the fixissue4052branch for the new pull request, or create a new branch?

**Comment by 1ucian0** [comment]
You could continue the same branch. However, it's good practice to start a new branch from master. You can move the changes you made in `fixissue4052branch` with `git merge fixissue4052branch`.

**Comment by enavarro51** [comment]
Thanks, @1ucian0. That's what I thought was a better plan.

**Pull request merged**